kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/settings/SnapshotGenericModal.java

```diff
@@ -0,0 +1,333 @@
+import com.google.gwt.event.dom.client.ClickEvent;
+import com.google.gwt.event.dom.client.KeyUpEvent;
+import com.google.gwt.event.shared.HandlerRegistration;
+import com.google.gwt.uibinder.client.UiBinder;
+import com.google.gwt.uibinder.client.UiField;
+import com.google.gwt.user.client.ui.Composite;
+import com.google.gwt.user.client.ui.Label;
+import com.google.gwt.user.client.ui.ScrollPanel;
+import com.google.gwt.user.client.ui.VerticalPanel;
+import com.google.gwt.user.client.ui.Widget;
+
+public abstract class SnapshotGenericModal extends Composite {
+
+    protected static final Messages MSGS = GWT.create(Messages.class);
+    private static SnapshotGenericModalUiBinder uiBinder = GWT.create(SnapshotGenericModalUiBinder.class);
+
+    interface SnapshotGenericModalUiBinder extends UiBinder<Widget, SnapshotGenericModal> {
+    }
+
+    @UiField
+    Modal snapshotModal;
+    @UiField
+    Form snapshotForm;
+    @UiField
+    Paragraph snapshotModalDescription;
+    @UiField
+    Paragraph snapshotModalHint;
+    @UiField
+    Label searchBoxSeparatorSmall;
+    @UiField
+    TextBox pidSearch;
+    @UiField
+    Label searchBoxSeparatorBig;
+    @UiField
+    ScrollPanel pidSelectionScrollPanel;
+    @UiField
+    Anchor selectOrRemoveAllAnchor;
+    @UiField
+    FormLabel noPidSelectedError;
+    @UiField
+    Label selectedPidCounter;
+    @UiField
+    ModalFooter snapshotFooter;
+
+    HandlerRegistration anchorClickHandler;
+
+    VerticalPanel pidPanel = new VerticalPanel();
+
+    protected SnapshotGenericModal() {
+        initWidget(uiBinder.createAndBindUi(this));
+        this.noPidSelectedError.setVisible(false);
+    }
+
+    /*
+     * Abstract initializers
+     */
+
+    /**
+     * Initialize the modal title, the description and the hint paragraphs of the modal.
+     */
+    abstract void initTitleAndDescriptionAndHint();
+
+    /**
+     * Initialize eventual buttons to perform the desired operations on the PIDs list.
+     * 
+     * @param snapshotId
+     *            - snapshotId of the snapshot from which the PIDs list was taken as {@link Long}
+     */
+    abstract void initEventButtons(Long snapshotId);
+
+    /**
+     * Initialize the footer of the modal. It can be used to add the buttons to perform the operations.
+     */
+    abstract void initFooter();
+
+    /**
+     * Initialize the eventual hidden field required to perform HTTP requests.
+     */
+    abstract void initHiddenFields();
+
+    /**
+     * Method used to initialize the required elements and eventually customize the modal.
+     * 
+     * To finally show the modal, the user must call 'this.snapshotModal.show()'.
+     * 
+     * @param snapshotId
+     *            - snapshotId to be downloaded as {@link Long}
+     * @param snapshotConfigs
+     *            - list of configuration contained in the snapshot to be downloaded as {@link List<String>}
+     */
+    abstract void show(Long snapshotId, List<String> snapshotConfigs);
```

Comment:
This can be put as not abstract. The code should be the same for each implementation.